### PR TITLE
EditorGUILayoutHelpers: Draw Options Selector

### PR DESCRIPTION
**Pull request by kkjamie**
Now you can draw a dropwdown list of options similar to what you get with an enum, but it's a string. Sometimes you have a list of things you grab at runtime, and can't possibly be an enum. Even at compile time I find string better than enums for serializing stuff since you don't have to worry about order.

Also worth noting I made the class partial and split by feature, renamed the original helpers file. and fixed a compiler warning

### CODE REVIEW

File: Utils/Editor/EditorGUIHelpers/DrawFieldByTypes.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ public static object FieldByType(string label, object obj, Type type)
 			// special case for object fields
 			if (typeof(UnityEngine.Object).IsAssignableFrom(type))
 			{
-				return EditorGUILayout.ObjectField(label, (UnityEngine.Object) obj, type);
+				return EditorGUILayout.ObjectField(label, (UnityEngine.Object) obj, type, true);
```
Comment (kkjamie) [inline Utils/Editor/EditorGUIHelpers/DrawFieldByTypes.cs]: This fixed a compiler warning about using an obsolete overload of this function

File: Utils/Editor/EditorGUIHelpers/OptionSelector.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+using System;
+using UnityEditor;
+
+namespace DUCK.Utils.Editor.EditorGUIHelpers
+{
+	public static partial class EditorGUILayoutHelpers
+	{
+		public static string OptionSelectorField(string label, string value, string[] options)
```
Comment (kkjamie) [inline Utils/Editor/EditorGUIHelpers/OptionSelector.cs]: At the moment if you pass in a value not found in the options it defaults to empty string, I guess we could provide a default value for such cases,

**Comment by Rollo-Liu** [comment]
Looking good!

File: Utils/Editor/EditorGUIHelpers/OptionSelector.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+		{
+			var index = Array.IndexOf(options, value);
+			var selectedIndex = EditorGUILayout.Popup(label, index, options);
+			return selectedIndex >= 0 ? options[selectedIndex] : "";
```
Comment (HarryHodgson) [inline Utils/Editor/EditorGUIHelpers/OptionSelector.cs]: I prefer `string.Empty` over `""`

**Pull request merged**